transhumandesign/kag-base
# [added] Resupply indicator support for archer & Sandbox mode

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

```
[added] Resupply indicator for Archer
[added] Resupply indicators to Sandbox mode
```

Fixes https://github.com/transhumandesign/kag-base/issues/1810

Screenshot: https://i.imgur.com/69ceaWf.png

Prior to this PR, resupply indicator only worked for Builders in CTF and small CTF.

This PR adds the resupply indicator icon to the HUD for archers, telling them how long until the next resupply of 30 arrows.
Resupply indicators are now shown not only in CTF/SmallCTF, but also in Sandbox.

I added both `warmup_arrows_amount` and `matchtime_arrows_amount` to `CTF_Common.as`, both valued 30.
I did this just to be in line with the existing variables. You could do it with one variable only.
Just tell me if you want it changed.

## Steps to Test or Reproduce

Go to different game modes and see the resupply icon being there when in CTF/SmallCTF/Sandbox and when playing as Archer or Builder.
